os/pm/pm_idle.c

```diff
@@ -164,13 +163,14 @@ void pm_idle(void)
 		/* Send signal to shutdown other cores here */
 		for (cpu = 1; cpu < CONFIG_SMP_NCPUS; cpu++) {
 			if (up_get_cpu_state(cpu) == CPU_RUNNING) {
+				/* Reset core gating status flag */
+				up_set_gating_flag_status(cpu, 0);
 				up_cpu_hotplug(cpu);
 			}
-			/* Reset core gating status flag */
-			up_set_gating_flag_status(cpu, 0);
```

Comment:
You cannot move this line above hot plug command, the reason of such design is to ensure that the gated core can handle the pending hot plug request when it has been granted access to exit the gated condition.
If you restore the gated core first, it will be possible that a context switch happened immediately before the hot plug request is sent out and received by the target core(ie. due to timing issue), then the target core will enter hot plug state but it is not in idle thread.
Please revert this change.

Reply:
`pm_idle` is inside `critical_section`. The `tick interrupts` are disabled at this time for all CPU cores. The context switch won't happen. Also, can you confirm, do we really need to gate CPUs inside critical section? Because all CPUs will not change their context inside critical section, and we are doing gating to ensure CPU did not change context before we check all CPUs are idle and ready to put into sleep.

Reply:
Critical section only disable interrupt as per each specific cores. pm_idle will only be executed by primary core. The generic clock for CA32 is disabled in the BSP layer which is invoked in the last step of pm_idle. Meaning, secondary cores are still possible to happen context switch when they exit from gated condition.
I don't understand your intention of swapping the sequence, but it shouldn't be allowed.

Reply:
done